core/src/main/java/org/testcontainers/DockerClientFactory.java

```diff
@@ -330,8 +329,17 @@ private boolean checkMountableFile() {
     public void checkAndPullImage(DockerClient client, String image) {
         try {
             client.inspectImageCmd(image).exec();
-        } catch (NotFoundException e) {
-            client.pullImageCmd(image).exec(new TimeLimitedLoggedPullImageResultCallback(log)).awaitCompletion();
+        } catch (NotFoundException notFoundException) {
+            PullImageCmd pullImageCmd = client.pullImageCmd(image);
+            try {
+                pullImageCmd.exec(new TimeLimitedLoggedPullImageResultCallback(log)).awaitCompletion();
+            } catch (DockerClientException e) {
+                // Try to fallback to x86
```

Comment:
Should we log that we're doing this?

Reply:
Good idea - will add 👍

Reply:
On a second thought (and a round of discussions on Slack), logging this would require a bit more work, so let's keep it as it is right now and do a follow up later 😅